triples-generator/src/test/scala/io/renku/triplesgenerator/config/VersionCompatibilityConfigSpec.scala

```diff
@@ -39,88 +37,63 @@ class VersionCompatibilityConfigSpec extends AnyWordSpec with should.Matchers {
 
     "fail if there are not value set for the matrix" in new TestCase {
 
-      val config = ConfigFactory.parseMap(
-        Map("compatibility-matrix" -> List.empty[String].asJava).asJava
-      )
-      val Failure(exception) = readVersionPairs(config)
+      val config             = ConfigFactory.parseString("compatibility { }")
+      val Failure(exception) = readCompatibilityConfig(config)
 
       exception            shouldBe a[Exception]
-      exception.getMessage shouldBe "No compatibility matrix provided for schema version"
+      exception.getMessage shouldBe "String: 1: No configuration setting found for key 'cli-version'"
     }
 
-    "return a list of VersionSchemaPairs if the value is set for the matrix" in new TestCase {
-      val cliVersionNumbers    = cliVersions.generateNonEmptyList(2, 10)
-      val schemaVersionNumbers = projectSchemaVersions.generateNonEmptyList(2, 10)
-
-      val expected = cliVersionNumbers.toList
-        .zip(schemaVersionNumbers.toList)
-        .map { case (cliVersion, schemaVersion) => RenkuVersionPair(cliVersion, schemaVersion) }
+    "return a Compatibility config if the value is set" in new TestCase {
+      val c = compatibilityGen.generateOne.copy(renkuDevVersion = None)
 
-      val unparsedConfigElements = expected.map { case RenkuVersionPair(cliVersion, schemaVersion) =>
-        s"${cliVersion.value} -> ${schemaVersion.value}"
-      }.asJava
+      val unparsedConfigElements =
+        s"""
+           |compatibility {
+           |  cli-version = "${c.cliVersion.value}"
+           |  schema-version = "${c.schemaVersion.value}"
+           |  re-provisioning-needed = ${c.reProvisioningNeeded}
+           |}
+           |""".stripMargin
 
-      val config = ConfigFactory.parseMap(
-        Map("compatibility-matrix" -> unparsedConfigElements).asJava
-      )
+      val config = ConfigFactory.parseString(unparsedConfigElements)
 
-      readVersionPairs(config) shouldBe NonEmptyList.fromListUnsafe(expected).pure[Try]
+      readCompatibilityConfig(config) shouldBe c.pure[Try]
     }
 
     "return a list of RenkuVersionPairs with the first element of the RenkuDevVersion if it exists and log a warning" in new TestCase {
```

Comment:
Maybe we could adapt the name of the test so it doesn't mention the VersionPairs but the new CompatibilityConfig